src/main/java/com/mraof/minestuck/item/loot/conditions/LandTypeLootCondition.java

```diff
@@ -52,7 +59,7 @@ public boolean test(LootContext context)
 		LandTypePair aspects = LandTypePair.getTypes(level).orElse(null);
 		
 		if(aspects != null && (terrainTypes.contains(aspects.getTerrain()) || titleTypes.contains(aspects.getTitle())
-				|| terrainGroups.contains(aspects.getTerrain().getGroup()) || titleGroups.contains(aspects.getTitle().getGroup())))
+				|| terrainTag != null && aspects.getTerrain().is(terrainTag) || titleTag != null && aspects.getTitle().is(titleTag)))
```

Comment:
Feel free to ignore but we should probably split this into a couple booleans or something for easier reading. We have so many OR operators here!

Reply:
Added some newlines to improve readability. Does that work?

Reply:
yup!